compression/gzip_zinfo_test.go

```diff
@@ -37,6 +37,36 @@ func TestNewGzipZinfo(t *testing.T) {
 			checkpoints: []byte{},
 			expectError: true,
 		},
+		{
+			name:        "checkpoints with less than 'header size' bytes header should return error",
+			checkpoints: []byte{00},
+			expectError: true,
+		},
+		{
+			name: "checkpoints with too few checkpoints should return error",
```

Comment:
nit: I feel `checkpoints with too few checkpoints` (and `checkpoints with zero checkpoints`) reads confusing. I guess the first `checkpoints` means `zinfoBytes` passed to `NewGzipZinfo` and the second `checkpoints` means `checkpoints` within the zinfo(Bytes). We probably can change the arg name `func NewGzipZinfo(checkpoints []byte)` to `func NewGzipZinfo(zinfoBytes []byte)`.